src/psmoveapi.cpp

```diff
@@ -151,11 +151,15 @@ PSMoveAPI::PSMoveAPI(EventReceiver *receiver, void *user_data)
     for (int i=0; i<n; i++) {
         PSMove *move = psmove_connect_by_id(i);
 
-        char *tmp = psmove_get_serial(move);
-        std::string serial(tmp);
-        free(tmp);
+        if (move) {
```

Comment:
Should we investigate why this happens and/or at least print a warning or something? Otherwise these errors will just be silent, and it will be harder to debug why it doesn't work?

Reply:
Fair point. Here is the call stack of my failure:
```
psmove_connect_by_id
psmove_get_serial
_psmove_read_btaddrs
hid_get_feature_report
```
Ultimately, [`DeviceIoControl`](https://github.com/thp/hidapi/blob/38681c734965636c9fd710dc3898d90aaf923bb9/windows/hid.c#L778-L794) was giving me unusual results. It would return successful, but the `bytes_returned` value would be 168 (after expecting 16) and `data` would be garbage. The [documentation ](https://docs.microsoft.com/en-us/windows/desktop/api/ioapiset/nf-ioapiset-deviceiocontrol)doesn't describe this scenario, if the output buffer is too small, the call should fail.  Also, if I just kept trying in a loop (or restarting the program), it would work eventually with `bytes_returned` 

After some trial and error, I discovered that only the ZCM2 models were exhibiting this behavior. It seems that these models want at least 20 bytes as opposed to ZCM1's 16. Once I changed this, it works every time.

I don't have a Linux or Mac build but I'm guessing they aren't exhibiting the same problem? Would you want an `#ifdef` path for Windows or port function? Or change the error checking behavior to not be so strict about bytes returned vs bytes expected?

Reply:
The HID descriptor for this report is identical on ZCM1 and ZCM2. Says it is 15 bytes (plus 1 for the report ID). More importantly, I have never seen hid_get_feature_report() return the wrong value for this particular report on Windows. Neither for ZCM1 nor ZCM2. Same on Linux. I have tested this with the example programs and tools that come with psmoveapi (such as examples/c/example.c). I currently cannot test on Windows 10, though, only older versions.

Could you please provide a minimal test program that triggers this behaviour? What version of Windows are you using? What is the exact controller model you are testing with?

Reply:
Sorry, I just realized this is not the right place to continue debugging this. Please open a ticket for this. We should continue there.

The actual code change, if it is supposed to stay, does not break anything, so it is probably fine. I agree that this should at least trigger a warning. I would, however, very much prefer investigating this issue further instead of just sticking a band-aid on it.

Reply:
Opened issue #378.

Reply:
Underlying issue was fixed.